django/db/backends/postgresql/base.py

```diff
@@ -364,21 +346,40 @@ def ensure_timezone(self):
         self.close_pool()
         if self.connection is None:
             return False
-        return ensure_timezone(self.connection, self.ops, self.timezone_name)
+        return self._configure_timezone(self.connection)
+
+    def ensure_role(self):
```

Comment:
I am not sure we want/should introduce that function. Overriding is possible via `_configure_role` anyways and to keep the patch small as well I wouldn't introduce new methods.

Reply:
Thank you @apollo13! I'll rebase onto main, fix conflicts, squash and push.

FYI: I restored `ensure_role` since it was available(ish) before the connection pool branch was merged: it felt like the right thing to do given the timezone "revert" we are sort of doing. I'm happy to remove though.